### PR TITLE
Pass cancellation token to Dequeue method, and rewrite comment

### PR DESCRIPTION
**Pull request by davidwengier**
If our server got shut down by the client, and we happened to be waiting to dequeue the next item from the queue, we would log it as an error because it came from an unknown cancellation token. This correctly links everything up.

Then I updated the comment because who understands Past Dave™

### CODE REVIEW

**Review by sharwell — APPROVED** [review]
I'm assuming we are sure `DequeueAsync` processes the `_cancelSource` cancellation before whatever other cancellation was involved to lead to the log.

File: src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs [hunk 1]
```diff
@@ -230,8 +230,10 @@ private async Task ProcessQueueAsync()
             }
             catch (OperationCanceledException e) when (e.CancellationToken == _cancelSource.Token)
```
Comment (sharwell) [inline src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs:231]: 💡 The other way to do this is:

```suggestion
            catch (OperationCanceledException) when (_cancelSource.IsCancellationRequested)
```

**Comment by davidwengier** [comment]
> I'm assuming we are sure DequeueAsync processes the _cancelSource cancellation before whatever other cancellation was involved to lead to the log.

I'm not sure exactly what you mean, but there is no "other" cancellation. The log happened because of the call to `_queue.Complete()` cancels the pending `DequeueAsync` call, but without a cancellation token in the exception. Or at least not the one we were checking for.

**Comment by sharwell** [comment]
> I'm not sure exactly what you mean, but there is no "other" cancellation.

The call to `_queue.Complete()` calls `TrySetCanceled()` on any remaining `DequeueAsync` waiters. It's important to ensure that all of those waiters have called `TrySetCanceled(token)` prior to the call to `Complete()` so the right cancellation token makes it all the way through to the location where you checked it.

**Pull request merged**